### PR TITLE
[BC-Breaking]Remove capture_pre_autograd_graph references in quantization

### PR DESCRIPTION
**Pull request by yushangdi**
Summary:
As title

This is a BC-breaking change because graph produced by "capture_pre_autograd_graph" cannot be input to quantization anymore. But this is ok, since this API is deprecated for a while and is going to be deleted. We have removed all call sites of it. 

We remove the deprecated API references in code, docs, and tests.

We also removed two tests that specific to capture_pre_autograd_graph API.

Test Plan: CI

Differential Revision: D65351887

cc @ezyang @gchanan

### CODE REVIEW

**Comment by jerryzh168** [comment]
cc @andrewor14 can you follow up on how does the change of training ir affect batchnorm op consolidation work?

**Comment by yushangdi** [comment]
There are some test failures because a call to `capture_pre_autograd_graph` is recently added in https://github.com/pytorch/pytorch/pull/139054. I'd have to wait until it's imported internally to overwrite it.

**Comment by andrewor14** [comment]
> cc @andrewor14 can you follow up on how does the change of training ir affect batchnorm op consolidation work?

Sorry just saw this. It actually makes batch norm consolidation much lower priority, because export no longer produces the same batch norm ops as before. In particular, now we get `aten.batch_norm` directly in the graph, which is kind of already the "consolidated" op from export perspective. Users won't see `aten.cudnn_batch_norm` or `aten._native_batch_norm_legit` anymore if they use `export_for_training`

**Comment by yushangdi** [comment]
@andrewor14 @jerryzh168 We've removed all "capture_pre_autograd_graph" call sites now, and we are finally deleting it. Can I get a review for this PR? Thanks!!

**Review by tugsbayasgalan — APPROVED** [review]

File: torch/ao/quantization/pt2e/utils.py [hunk 1]
```diff
@@ -381,10 +380,9 @@ def _get_aten_graph_module_for_pattern(
             kwargs,
         ).module()
     else:
-        aten_pattern = capture_pre_autograd_graph(
-            pattern,  # type: ignore[arg-type]
-            example_inputs,
-            kwargs,
+        raise RuntimeError(
+            "capture_pre_autograd_graph is deprecated and will be deleted soon."
+            "Please use torch.export.export_for_training instead."
```
Comment (andrewor14) [inline torch/ao/quantization/pt2e/utils.py:385]: This is a BC-breaking change right? Should we mark it as such in the PR title and description?

**Review by andrewor14 — APPROVED** [review]
Looks great, thanks! Please also import this to make sure no existing use cases are being affected.

File: torch/ao/quantization/pt2e/utils.py (hunk 1 above)
Comment (yushangdi) [reply torch/ao/quantization/pt2e/utils.py:385]: > This is a BC-breaking change right? Should we mark it as such in the PR title and description?

sure!

**Review by jerryzh168 — APPROVED** [review]
looks good, @andrewor14 can you follow up on the batchnorm and qat related things? (maybe make a post?) I'm still not clear on how the capture changes these things
